datalad/distribution/get.py

```diff
@@ -77,36 +77,99 @@
 lgr = logging.getLogger('datalad.distribution.get')
 
 
-def _get_flexible_source_candidates_for_submodule(ds, sm_path, sm_url=None):
-    """Retrieve candidates from where to install the submodule
+def _get_remotes_having_commit(repo, commit_hexsha, with_urls_only=True):
+    """Traverse all branches of the remote and check if commit in any of their ancestry
 
-    Even if url for submodule is provided explicitly -- first tries urls under
-    parent's module tracking branch remote.
+    It is a generator yielding names of the remotes
     """
-    clone_urls = []
+    remote_branches = [
+        b['refname:strip=2']
+        for b in repo.for_each_ref_(
+            fields='refname:strip=2',
+            pattern='refs/remotes',
+            contains=commit_hexsha)]
+    return [
+        remote
+        for remote in repo.get_remotes(with_urls_only=with_urls_only)
+        if any(rb.startswith(remote + '/') for rb in remote_branches)
```

Comment:
I'm not sure I see the point of using `get_remote(with_urls_only=...)` here, especially given this is only used in one spot.  Wouldn't `for_each_ref_`s output (with an additional layer stripped) be sufficient?

Reply:
This is just keeping what the keep did before. I tried to get rid of it, but couldn't figure out why things would break. I'll leave it for another time.

Reply:
> This is just keeping what the keep did before.

Yes...

> I tried to get rid of it, but couldn't figure out why things would break.

ah, all right, so my guess that it isn't need is wrong.  Thanks.